azure-kusto-data/azure/kusto/data/kcsb.py

```diff
@@ -1,10 +1,102 @@
+import dataclasses
 from enum import unique, Enum
 from typing import Union, Callable, Coroutine, Optional, Tuple, List, Any
 from urllib.parse import urlparse
 
 from ._string_utils import assert_string_is_not_empty
 from ._token_providers import DeviceCallbackType
 from .client_details import ClientDetails
+from .helpers import load_bundled_json
+
+
+@unique
+class ValidKeywords(Enum):
+    DATA_SOURCE = "Data Source"
+    INITIAL_CATALOG = "Initial Catalog"
+    FEDERATED_SECURITY = "AAD Federated Security"
+    APPLICATION_CLIENT_ID = "Application Client Id"
+    APPLICATION_KEY = "Application Key"
+    USER_ID = "User ID"
+    PASSWORD = "Password"
+    AUTHORITY_ID = "Authority Id"
+    APPLICATION_TOKEN = "Application Token"
+    USER_TOKEN = "User Token"
+    APPLICATION_CERTIFICATE_BLOB = "Application Certificate Blob"
+    APPLICATION_CERTIFICATE_X5C = "Application Certificate SendX5c"
+    APPLICATION_CERTIFICATE_THUMBPRINT = "Application Certificate Thumbprint"
+    TRACE_APP_NAME = "Application Name for Tracing"
+    TRACE_USER_NAME = "User Name for Tracing"
+
+
+valid_keywords = [k.value for k in ValidKeywords]
+
+
+@unique
+class InvalidKeywords(Enum):
+    DSTS_FEDERATED_SECURITY = "dSTS Federated Security"
+    STREAMING = "Streaming"
+    UNCOMPRESSED = "Uncompressed"
+    ENFORCE_MFA = "EnforceMfa"
+    ACCEPT = "Accept"
+    QUERY_CONSISTENCY = "Query Consistency"
+    PASSWORD = "Password"
+    DATA_SOURCE_URI = "Data Source Uri"
+    AZURE_REGION = "Azure Region"
+    NAMESPACE = "Namespace"
+    APPLICATION_CERTIFICATE_ISSUER_DISTINGUISHED_NAME = "Application Certificate Issuer Distinguished Name"
+    APPLICATION_CERTIFICATE_SUBJECT_DISTINGUISHED_NAME = "Application Certificate Subject Distinguished Name"
+
+
+invalid_keywords = [k.value for k in InvalidKeywords]
+
+
+@dataclasses.dataclass(frozen=True)
+class Keyword:
+    name: ValidKeywords
+    aliases: List[str]
+    type: str
+    secret: bool
+
+    def is_str_type(self) -> bool:
+        return self.type == "string"
+
+    def is_bool_type(self) -> bool:
+        return self.type == "bool"
+
+
+kcsb_json: dict = load_bundled_json("kcsb.json")
```

Comment:
Can this be in the "static" init of some class

Reply:
Like this?